structs.go

```diff
@@ -286,25 +294,41 @@ type Channel struct {
 	// The user limit of the voice channel.
 	UserLimit int `json:"user_limit"`
 
-	// The ID of the parent channel, if the channel is under a category
+	// The ID of the parent channel, if the channel is under a category. For threads - id of the channel thread was created in.
 	ParentID string `json:"parent_id"`
 
-	// Amount of seconds a user has to wait before sending another message (0-21600)
+	// Amount of seconds a user has to wait before sending another message or creating another thread (0-21600)
 	// bots, as well as users with the permission manage_messages or manage_channel, are unaffected
 	RateLimitPerUser int `json:"rate_limit_per_user"`
 
-	// ID of the DM creator Zeroed if guild channel
+	// ID of the creator of the group DM or thread
 	OwnerID string `json:"owner_id"`
 
 	// ApplicationID of the DM creator Zeroed if guild channel or not a bot user
 	ApplicationID string `json:"application_id"`
+
+	// Thread-specific fields not needed by other channels
+	ThreadMetadata *ThreadMetadata `json:"thread_metadata,omitempty"`
+	// Thread member object for the current user, if they have joined the thread, only included on certain API endpoints
+	Member *ThreadMember `json:"thread_member"`
+
+	// All thread members. State channels only.
+	Members []*ThreadMember `json:"-"`
 }
 
 // Mention returns a string which mentions the channel
 func (c *Channel) Mention() string {
 	return fmt.Sprintf("<#%s>", c.ID)
 }
 
+// Thread is a subchannel inside an existing channel
+type Thread = Channel
```

Comment:
I don't really see the point of this type alias. It seems like it's only going to confuse terminology further—knowing that a thread is simply a type of channel is, IMO, necessary to using them in many ways (e.g. knowing how to send a message to one). Below, the use of the alias in `IsThread` also is somewhat confusing—why would a method called `IsThread` on a receiver of type `Thread` return false? I'd just stick with `Channel`.